eng/Tools.props

```diff
@@ -59,14 +59,24 @@
     <RepoTool Include="dotnet-reportgenerator-globaltool" Version="$(DotNetReportGeneratorGlobalToolPackageVersion)" />
   </ItemGroup>
 
-  <!--
-    Temporarily opt-in as dotnet restore interactive doesn't seem to work.
-    https://github.com/NuGet/Home/issues/7812. Use nuget restore meanwhile.
-  -->
+  <!-- Opt-in target to restore internal tools with interactive authentication. -->
   <Target Name="InitOptionalTools"
           Condition="'$(RestoreInternalTools)' == 'true'"
           AfterTargets="Restore">
 
+    <PropertyGroup>
+      <ToolsDir>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', '.tools', 'CredentialsProvider'))</ToolsDir>
+      <CredentialsProviderScriptUrl Condition="'$(OS)' == 'Windows_NT'">https://raw.githubusercontent.com/Microsoft/artifacts-credprovider/master/helpers/installcredprovider.ps1</CredentialsProviderScriptUrl>
```

Comment:
Maybe the base URL could be moved to a common property, so that if needs to be updated later we only need to update it in one place.

Reply:
That's over-engineering :D

Reply:
😢 ok